templates/web-ingress.yaml

```diff
@@ -20,9 +20,12 @@ metadata:
 spec:
   rules:
     {{- range .Values.web.ingress.hosts }}
-    - host: {{ . }}
+    - host: {{ if .name }}{{ .name }}{{ else }}{{ . }}{{ end }}
```

Comment:
Hmmm.
I tried with reflections, ultimately the template needs to call `.name` or `get "name" .`. The former fails even when guarded by `if not (isKind "string" .)` because the Go template eagerly attempts to render `.name` anyway, and the latter fails because `get` requires a `map[string]string` to be provided as an argument, which a `string` isn't, even when attempting to use reflection to prevent the `get` call because, again, it's eagerly attempted regardless of the `if / else`. It seems like `hosts` will be a list of strings, and that's it.

I don't really like it, but what do you think about something like `web.ingress.complexHosts: {}`?

Reply:
That's a shame. I'm fine with adding `web.ingress.complexHosts`, safest thing we can do in this instance I think.